pkg/executor/statefulset.go

```diff
@@ -252,7 +252,7 @@ func getStatefulset(
 		}
 	}
 
-	useServiceAccount := false
+	useServiceAccount := true
```

Comment:
What if serviceAccountName is nil or empty?

Reply:
k8s allows automountServiceAccountToken to be true without serviceAccountName. So we shouldn't need explicit validation.

Reply:
Kubernetes will mount the default service account if automountServiceAccountToken is set (which is also the default) and serviceAccountName is not specified. That may not be the intended behavior. Why surface the service account if not requested. We should continue to set automountServiceAccountToken to false if service account is not provided through cluster cr

Reply:
Hmm, yeah makes sense.